crates/ruff/src/rules/ruff/rules/blank_comment.rs

```diff
@@ -0,0 +1,63 @@
+use crate::registry::Rule;
+use crate::settings::{flags, Settings};
+use once_cell::sync::Lazy;
+use regex::Regex;
+use ruff_diagnostics::{AlwaysAutofixableViolation, Diagnostic, Edit};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::newlines::Line;
+use ruff_text_size::{TextLen, TextRange, TextSize};
+
+/// ## What it does
+/// Check for blank comments.
+///
+/// ## Why is this bad?
+/// Blank comments are useless and should be removed.
+///
+/// ## Example
+/// ```python
+/// print("Hello, World!")  #
+/// ```
+///
+/// Use instead:
+/// ```python
+/// print("Hello, World!")
+/// ```
+///
+/// ## References
+/// - [Ruff documentation](https://beta.ruff.rs/docs/configuration/#error-suppression)
+#[violation]
+pub struct BlankComment;
+
+impl AlwaysAutofixableViolation for BlankComment {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Blank comments are useless and should be removed")
+    }
+
+    fn autofix_title(&self) -> String {
+        "Remove blank comment".to_string()
+    }
+}
+
+static BLACK_COMMENT_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"\S(\s*#\s*)$").unwrap());
+
+/// RUF010
+pub fn blank_comment(
+    diagnostics: &mut Vec<Diagnostic>,
+    line: &Line,
+    settings: &Settings,
+    autofix: flags::Autofix,
+) {
+    if let Some(captures) = BLACK_COMMENT_REGEX.captures(line.as_str()) {
```

Comment:
Ruff has the `Indexer` data structure that exposes the range of all comments (`Indexer::comment_ranges`). You can use this with `Locator.slice(range)` to retrieve all comments and call the rule at the top level (module) once. This should be faster than running a regex on every line. 

I think we should also be able to replace the regex all together by manually skipping the `#` and then testing if `text.trim().is_empty()`.

Reply:
Yeah I guess we could do something like this

```
// Retrieve the comment using Locator.slice
        let comment = Locator.slice(range);

        // Skip the # at the start of the comment and trim white spaces
        let trimmed_comment = comment.strip_prefix("#").unwrap_or("").trim();

        // If the trimmed comment is empty, it's a blank comment
        if trimmed_comment.is_empty() {
            let mut diagnostic = Diagnostic::new(BlankComment, range);
            if autofix.into() && settings.rules.should_fix(Rule::BlankComment) {
                diagnostic.set_fix(Edit::deletion(range.start(), range.end()));
            }
            diagnostics.push(diagnostic);
        }
    }
    
 ```